crates/contract/src/call.rs

```diff
@@ -454,10 +465,21 @@ impl<P, D: CallDecoder> std::fmt::Debug for CallBuilder<P, D> {
 #[allow(unused_imports)]
 mod tests {
     use super::*;
+    use alloy_network::Ethereum;
     use alloy_node_bindings::{Anvil, AnvilInstance};
     use alloy_primitives::{address, b256, bytes, hex};
-    use alloy_providers::tmp::{HttpProvider, Provider};
+    use alloy_providers::{HttpProvider, Provider, RootProvider};
+    use alloy_rpc_client::RpcClient;
     use alloy_sol_types::sol;
+    use alloy_transport_http::Http;
+    use reqwest::Client;
+
+    fn spawn_anvil() -> (HttpProvider<Ethereum>, AnvilInstance) {
+        let anvil = Anvil::new().spawn();
+        let url = anvil.endpoint().parse().unwrap();
+        let http = Http::<Client>::new(url);
+        (RootProvider::<Ethereum, _>::new(RpcClient::new(http, true)), anvil)
```

Comment:
follow-on we need `Provider::<Ethereum>::connect(url)` which figures out which BoxTransport to use based on URL prefix ("HTTP://" vs "ws://" vs "ipc path")